paymentsheet/src/main/java/com/stripe/android/paymentsheet/PaymentSheetViewModel.kt

```diff
@@ -81,7 +81,7 @@ internal class PaymentSheetViewModel @Inject internal constructor(
     @IOContext workContext: CoroutineContext,
     savedStateHandle: SavedStateHandle,
     linkHandler: LinkHandler,
-    intentConfirmationHandlerFactory: IntentConfirmationHandler.Factory,
+    defaultConfirmationHandlerFactory: DefaultConfirmationHandler.Factory,
```

Comment:
We'll probably need an interface for this `Factory` as well. The `viewModelScope` cannot be injected until after the view model is created therefore the view models would need to create the handler with the `viewModelScope`
```kotlin
internal interface ConfirmationHandler.Factory {
    fun create(scope: CoroutineScope): ConfirmationHandler
}
```